src/session/session_api.c

```diff
@@ -1465,10 +1471,32 @@ __wt_session_range_truncate(
         WT_ERR(__session_open_cursor((WT_SESSION *)session, NULL, stop, NULL, &debug_stop));
 #endif
 
-    WT_ERR(__wt_schema_range_truncate(session, start, stop));
+    WT_ERR(__wt_schema_range_truncate(session, start, stop, &is_col_fix));
 
 #ifdef HAVE_DIAGNOSTIC
     is_truncate = true;
+    /*
+     * The debug cursors will be positioned at the start and stop keys of the range if there is one.
+     * For row-store and variable-length column store, we expect a WT_NOTFOUND value when searching
+     * for a record that has been truncated. For fixed length column store, this works a little
+     * differently. We should instead check that the corresponding value of the truncated record is
+     * zero.
+     */
+    if (!is_col_fix) {
```

Comment:
As mentioned in a previous comment, you could cast a cursor to `WT_CURSOR_BTREE*` and then have access to the `btree->type` to check if it's a FCS.